solutions/go/init/code/dbfile/page.go

```diff
@@ -0,0 +1,25 @@
+package dbfile
+
+import "encoding/binary"
+
+type page []byte
+
+func makePage(size int64) page {
+	return make(page, size)
+}
+
+func (p page) u8(off int) uint8 {
```

Comment:
I see these differences here.

My names are too short. I can rename them u8 -> uint8 (as getters are [officially named like this](https://go.dev/doc/effective_go#Getters)) or u8 -> parseUint8.

My functions do not check for buffer size and do not return error. I want to make it impossible to fail here. All the checks are done in place of reading the `page`, so that in the rest of the code I could avoid error checking and have more compact code.

They also use constants, but I think it's totally excessive here as it doesn't add any understanding and doesn't prevent any bugs.

What do you think?

Reply:
encoding/binary package uses Uint16 by the way.
https://pkg.go.dev/encoding/binary#ByteOrder

Reply:
In the example, they wrapped byte array to have interface similar to bufio.Reader and os.File. We here have another aim, we want to reduce repetition and incapsulate some internal task if case we would change how we store the pages.

Reply:
I agree that it makes sense to keep these reading functions in a single place. Let me think about this a bit, will reply soon!